src/hotspot/os/posix/perfMemory_posix.cpp

```diff
@@ -527,6 +527,7 @@ static char* get_user_name_slow(int vmid, int nspid, TRAPS) {
   // directory search
   char* oldest_user = NULL;
   time_t oldest_ctime = 0;
+  char buffer[MAXPATHLEN + 1];
```

Comment:
If you move this inside the #if defined(LINUX) it would be better. No risk of perturbing other platforms then.